### PR TITLE
fix: Revert "fix: updates from review"

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Reverts CoffeaTeam/coffea#1130

### CODE REVIEW

File: src/coffea/nanoevents/methods/candidate.py [hunk 1]
```diff
@@ -71,29 +70,23 @@ class PtEtaPhiECandidate(Candidate, vector.PtEtaPhiELorentzVector):
...
-)
-register_projection_classes(
-    PtEtaPhiECandidateArray,  # noqa: F821
-    vector.PolarTwoVectorArray,
-    vector.SphericalThreeVectorArray,
-    vector.LorentzVectorArray,
-    PtEtaPhiECandidateArray,  # noqa: F821
+awkward.behavior.update(
+    awkward._util.copy_behaviors(vector.LorentzVector, Candidate, behavior)
 )
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/candidate.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(vector.LorentzVector, Candidate, behavior))
```

File: src/coffea/nanoevents/methods/delphes.py [hunk 2]
```diff
@@ -131,16 +130,15 @@ def z(self):
...
-
-register_projection_classes(
-    VertexArray,  # noqa: F821
-    vector.TwoVectorArray,  # noqa: F821
-    vector.ThreeVectorArray,  # noqa: F821
-    VertexArray,  # noqa: F821
-    vector.LorentzVectorArray,  # noqa: F821
+awkward.behavior.update(
+    awkward._util.copy_behaviors(vector.LorentzVector, Vertex, behavior)
 )
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/delphes.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(vector.LorentzVector, Vertex, behavior))
```

File: src/coffea/nanoevents/methods/delphes.py [hunk 3]
```diff
@@ -180,17 +178,14 @@ def mass(self):
 
 
 _set_repr_name("Particle")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/delphes.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/delphes.py [hunk 4]
```diff
@@ -201,95 +196,83 @@ def mass(self):
...
 _set_repr_name("MasslessParticle")
-behavior.update(awkward._util.copy_behaviors(Particle, MasslessParticle, behavior))
-
-register_projection_classes(
-    MasslessParticleArray,  # noqa: F821
-    vector.TwoVectorArray,  # noqa: F821
-    vector.ThreeVectorArray,  # noqa: F821
-    MasslessParticleArray,  # noqa: F821
-    vector.LorentzVectorArray,  # noqa: F821
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/delphes.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/delphes.py [hunk 5]
```diff
@@ -201,95 +196,83 @@ def mass(self):
...
 _set_repr_name("Photon")
-behavior.update(awkward._util.copy_behaviors(MasslessParticle, Photon, behavior))
-
-register_projection_classes(
-    PhotonArray,  # noqa: F821
-    vector.TwoVectorArray,  # noqa: F821
-    vector.ThreeVectorArray,  # noqa: F821
-    PhotonArray,  # noqa: F821
-    vector.LorentzVectorArray,  # noqa: F821
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/delphes.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/delphes.py [hunk 6]
```diff
@@ -201,95 +196,83 @@ def mass(self):
...
 _set_repr_name("Electron")
-behavior.update(awkward._util.copy_behaviors(MasslessParticle, Electron, behavior))
-
-register_projection_classes(
-    ElectronArray,  # noqa: F821
-    vector.TwoVectorArray,  # noqa: F821
-    vector.ThreeVectorArray,  # noqa: F821
-    ElectronArray,  # noqa: F821
-    vector.LorentzVectorArray,  # noqa: F821
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/delphes.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/delphes.py [hunk 7]
```diff
@@ -201,95 +196,83 @@ def mass(self):
...
-behavior.update(awkward._util.copy_behaviors(MasslessParticle, Muon, behavior))
-
-register_projection_classes(
-    MuonArray,  # noqa: F821
-    vector.TwoVectorArray,  # noqa: F821
-    vector.ThreeVectorArray,  # noqa: F821
-    MuonArray,  # noqa: F821
-    vector.LorentzVectorArray,  # noqa: F821
-)
+awkward.behavior.update(awkward._util.copy_behaviors(MasslessParticle, Muon, behavior))
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/delphes.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(MasslessParticle, Muon, behavior))
```

File: src/coffea/nanoevents/methods/delphes.py [hunk 8]
```diff
@@ -201,95 +196,83 @@ def mass(self):
...
-behavior.update(awkward._util.copy_behaviors(Particle, Jet, behavior))
-
-register_projection_classes(
-    JetArray,  # noqa: F821
-    vector.TwoVectorArray,  # noqa: F821
-    vector.ThreeVectorArray,  # noqa: F821
-    JetArray,  # noqa: F821
-    vector.LorentzVectorArray,  # noqa: F821
-)
+awkward.behavior.update(awkward._util.copy_behaviors(Particle, Jet, behavior))
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/delphes.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(Particle, Jet, behavior))
```

File: src/coffea/nanoevents/methods/delphes.py [hunk 9]
```diff
@@ -201,95 +196,83 @@ def mass(self):
...
-behavior.update(awkward._util.copy_behaviors(Particle, Track, behavior))
-
-register_projection_classes(
-    TrackArray,  # noqa: F821
-    vector.TwoVectorArray,  # noqa: F821
-    vector.ThreeVectorArray,  # noqa: F821
-    TrackArray,  # noqa: F821
-    vector.LorentzVectorArray,  # noqa: F821
-)
+awkward.behavior.update(awkward._util.copy_behaviors(Particle, Track, behavior))
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/delphes.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(Particle, Track, behavior))
```

File: src/coffea/nanoevents/methods/delphes.py [hunk 10]
```diff
@@ -300,15 +283,12 @@ def pt(self):
...
-behavior.update(awkward._util.copy_behaviors(MasslessParticle, Tower, behavior))
-
-register_projection_classes(
-    TowerArray,  # noqa: F821
-    vector.TwoVectorArray,  # noqa: F821
-    vector.ThreeVectorArray,  # noqa: F821
-    TowerArray,  # noqa: F821
-    vector.LorentzVectorArray,  # noqa: F821
-)
+awkward.behavior.update(awkward._util.copy_behaviors(MasslessParticle, Tower, behavior))
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/delphes.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(MasslessParticle, Tower, behavior))
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 11]
```diff
@@ -39,19 +38,16 @@ class PtEtaPhiMCollection(vector.PtEtaPhiMLorentzVector, base.NanoCollection):
     pass
 
 
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 12]
```diff
@@ -160,17 +156,14 @@ def distinctChildrenDeep(self, dask_array):
 
 
 _set_repr_name("GenParticle")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 13]
```diff
@@ -191,17 +184,14 @@ def parent(self, dask_array):
 
 
 _set_repr_name("GenVisTau")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 14]
```diff
@@ -266,17 +256,14 @@ def matched_photon(self, dask_array):
 
 
 _set_repr_name("Electron")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 15]
```diff
@@ -353,17 +340,14 @@ def matched_jet(self, dask_array):
 
 
 _set_repr_name("Muon")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 16]
```diff
@@ -388,17 +372,14 @@ def matched_jet(self, dask_array):
 
 
 _set_repr_name("Tau")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 17]
```diff
@@ -474,17 +455,14 @@ def matched_jet(self, dask_array):
 
 
 _set_repr_name("Photon")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 18]
```diff
@@ -501,17 +479,14 @@ def matched_muon(self, dask_array):
 
 
 _set_repr_name("FsrPhoton")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 19]
```diff
@@ -582,17 +557,14 @@ def constituents(self, dask_array):
 
 
 _set_repr_name("Jet")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 20]
```diff
@@ -655,17 +627,14 @@ def constituents(self, dask_array):
 
 
 _set_repr_name("FatJet")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/nanoaod.py [hunk 21]
```diff
@@ -678,17 +647,14 @@ def r(self):
 
 
 _set_repr_name("MissingET")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/nanoaod.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/pdune.py [hunk 22]
```diff
@@ -93,16 +92,15 @@ def mass(self):
...
-behavior.update(awkward._util.copy_behaviors(vector.LorentzVector, Particle, behavior))
-
-register_projection_classes(
-    ParticleArray,  # noqa: F821
-    vector.TwoVectorArray,
-    vector.ThreeVectorArray,
-    ParticleArray,  # noqa: F821
-    vector.LorentzVectorArray,
+awkward.behavior.update(
+    awkward._util.copy_behaviors(vector.LorentzVector, Particle, behavior)
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/pdune.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(vector.LorentzVector, Particle, behavior))
```

File: src/coffea/nanoevents/methods/pdune.py [hunk 23]
```diff
@@ -93,16 +92,15 @@ def mass(self):
...
 )
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/pdune.py]:
```suggestion
```

File: src/coffea/nanoevents/methods/pdune.py [hunk 24]
```diff
@@ -140,17 +138,14 @@ def t(self):
 
 
 _set_repr_name("TrackParticle")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/pdune.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/pdune.py [hunk 25]
```diff
@@ -170,15 +165,12 @@ def trackParticle(self):
...
-behavior.update(awkward._util.copy_behaviors(Particle, Muon, behavior))
-
-register_projection_classes(
-    MuonArray,  # noqa: F821
-    vector.TwoVectorArray,
-    vector.ThreeVectorArray,
-    MuonArray,  # noqa: F821
-    vector.LorentzVectorArray,
-)
+awkward.behavior.update(awkward._util.copy_behaviors(Particle, Muon, behavior))
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/pdune.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(Particle, Muon, behavior))
```

File: src/coffea/nanoevents/methods/pdune.py [hunk 26]
```diff
@@ -206,15 +198,12 @@ def trackParticle(self):
...
-behavior.update(awkward._util.copy_behaviors(Particle, Electron, behavior))
-
-register_projection_classes(
-    ElectronArray,  # noqa: F821
-    vector.TwoVectorArray,
-    vector.ThreeVectorArray,
-    ElectronArray,  # noqa: F821
-    vector.LorentzVectorArray,
-)
+awkward.behavior.update(awkward._util.copy_behaviors(Particle, Electron, behavior))
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/pdune.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(Particle, Electron, behavior))
```

File: src/coffea/nanoevents/methods/pdune.py [hunk 27]
```diff
@@ -257,14 +246,11 @@ def parents(self):
 
 
 _set_repr_name("TruthParticle")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/pdune.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/physlite.py [hunk 28]
```diff
@@ -144,17 +143,14 @@ def mass(self):
 
 
 _set_repr_name("Particle")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/physlite.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/physlite.py [hunk 29]
```diff
@@ -193,17 +189,14 @@ def t(self):
 
 
 _set_repr_name("TrackParticle")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/physlite.py]:
```suggestion
behavior.update(
```

File: src/coffea/nanoevents/methods/physlite.py [hunk 30]
```diff
@@ -232,15 +225,12 @@ def trackParticle(self, dask_array):
...
-behavior.update(awkward._util.copy_behaviors(Particle, Muon, behavior))
-
-register_projection_classes(
-    MuonArray,  # noqa: F821
-    vector.TwoVectorArray,
-    vector.ThreeVectorArray,
-    MuonArray,  # noqa: F821
-    vector.LorentzVectorArray,
-)
+awkward.behavior.update(awkward._util.copy_behaviors(Particle, Muon, behavior))
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/physlite.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(Particle, Muon, behavior))
```

File: src/coffea/nanoevents/methods/physlite.py [hunk 31]
```diff
@@ -293,15 +283,12 @@ def caloClusters(self, dask_array):
...
-behavior.update(awkward._util.copy_behaviors(Particle, Electron, behavior))
-
-register_projection_classes(
-    ElectronArray,  # noqa: F821
-    vector.TwoVectorArray,
-    vector.ThreeVectorArray,
-    ElectronArray,  # noqa: F821
-    vector.LorentzVectorArray,
-)
+awkward.behavior.update(awkward._util.copy_behaviors(Particle, Electron, behavior))
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/physlite.py]:
```suggestion
behavior.update(awkward._util.copy_behaviors(Particle, Electron, behavior))
```

File: src/coffea/nanoevents/methods/physlite.py [hunk 32]
```diff
@@ -344,14 +331,11 @@ def parents(self):
 
 
 _set_repr_name("TruthParticle")
-behavior.update(
+awkward.behavior.update(
```
Comment (Saransh-cpp) [inline src/coffea/nanoevents/methods/physlite.py]:
```suggestion
behavior.update(
```

File: tests/test_nanoevents_vector.py [hunk 33]
```diff
@@ -543,13 +543,6 @@ def test_inherited_method_transpose(lcoord, threecoord, twocoord):
     assert_record_arrays_equal(a.like(c) + c, c + a.like(c), check_type=True)
     assert_record_arrays_equal(b.like(c) + c, c + b.like(c), check_type=True)
```
Comment (Saransh-cpp) [inline tests/test_nanoevents_vector.py]:
```suggestion

with pytest.raises(TypeError):
         a + b == b + a
     with pytest.raises(TypeError):
         a + c == c + a
     with pytest.raises(TypeError):
         b + c == c + b

```

File: tests/test_nanoevents_vector.py [hunk 34]
```diff
@@ -558,13 +551,6 @@ def test_inherited_method_transpose(lcoord, threecoord, twocoord):
     assert_record_arrays_equal((a.like(c) - c), -(c - a.like(c)), check_type=True)
     assert_record_arrays_equal((b.like(c) - c), -(c - b.like(c)), check_type=True)
```
Comment (Saransh-cpp) [inline tests/test_nanoevents_vector.py]:
```suggestion

    with pytest.raises(TypeError):
         a - b == -(b - a)
     with pytest.raises(TypeError):
         a - c == -(c - a)
     with pytest.raises(TypeError):
         b - c == -(c - b)

```

**Review by Saransh-cpp — COMMENTED** [review]
Add back useful changes

File: tests/test_nanoevents_vector.py [hunk 35]
```diff
@@ -559,11 +559,11 @@ def test_inherited_method_transpose(lcoord, threecoord, twocoord):
...
-        a - b == -(b - a)
-    with pytest.raises(TypeError):
-        a - c == -(c - a)
-    with pytest.raises(TypeError):
-        b - c == -(c - b)
+         a - b == -(b - a)
+     with pytest.raises(TypeError):
+         a - c == -(c - a)
+     with pytest.raises(TypeError):
+         b - c == -(c - b)
```
Comment (Saransh-cpp) [inline tests/test_nanoevents_vector.py]:
```suggestion
        a - b == -(b - a)
    with pytest.raises(TypeError):
        a - c == -(c - a)
    with pytest.raises(TypeError):
        b - c == -(c - b)
```

File: tests/test_nanoevents_vector.py [hunk 36]
```diff
@@ -543,12 +543,12 @@ def test_inherited_method_transpose(lcoord, threecoord, twocoord):
...
-    with pytest.raises(TypeError):
-        a + c == c + a
-    with pytest.raises(TypeError):
-        b + c == c + b
+with pytest.raises(TypeError):
+         a + b == b + a
+     with pytest.raises(TypeError):
+         a + c == c + a
+     with pytest.raises(TypeError):
+         b + c == c + b
```
Comment (Saransh-cpp) [inline tests/test_nanoevents_vector.py]:
```suggestion
    with pytest.raises(TypeError):
        a + b == b + a
    with pytest.raises(TypeError):
        a + c == c + a
    with pytest.raises(TypeError):
        b + c == c + b
```

**Comment by lgray** [comment]
@Saransh-cpp this one is ready?

**Comment by Saransh-cpp** [comment]
Yes!

**Pull request merged**